server/eclipse-project/src/main/java/us/freeandfair/corla/model/CountyQueries.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Free & Fair Colorado RLA System
+ * 
+ * @title ColoradoRLA
+ * @created Aug 8, 2017
+ * @copyright 2017 Free & Fair
+ * @license GNU General Public License 3.0
+ * @author Daniel M. Zimmerman <dmz@galois.com>
+ * @description A system to assist in conducting statewide risk-limiting audits.
+ */
+
+package us.freeandfair.corla.model;
+
+import java.util.List;
+
+import javax.persistence.PersistenceException;
+import javax.persistence.RollbackException;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.hibernate.Session;
+
+import us.freeandfair.corla.Main;
+import us.freeandfair.corla.hibernate.Persistence;
+
+/**
+ * Queries having to do with County entities.
+ * 
+ * @author Daniel M. Zimmerman
+ * @version 0.0.1
+ */
+public final class CountyQueries {
+  /**
+   * Private constructor to prevent instantiation.
+   */
+  private CountyQueries() {
+    // do nothing
+  }
+  
+  /**
+   * Obtain the County object with a specific ID number, if one exists.
+   *
+   * @param the_id The ID number.
+   * @return the matched county, if one exists, or null otherwise.
+   */
+  // we are checking to see if exactly one result is in a list, and
+  // PMD doesn't like it
+  @SuppressWarnings("PMD.AvoidLiteralsInIfCondition")
+  public static County byID(final Integer the_id) {
+    County result = null;
+    
+    try {
+      final boolean transaction = Persistence.beginTransaction();
+      final Session s = Persistence.currentSession();
+      final CriteriaBuilder cb = s.getCriteriaBuilder();
+      final CriteriaQuery<County> cq = cb.createQuery(County.class);
+      final Root<County> root = cq.from(County.class);
+      cq.select(root).where(cb.equal(root.get("my_identifier"), the_id));
+      final TypedQuery<County> query = s.createQuery(cq);
+      final List<County> query_results = query.getResultList();
+      // if there's exactly one result, return that
+      if (query_results.size() == 1) {
+        result = query_results.get(0);
+      } 
+      if (transaction) {
```

Comment:
It would be really nice to encapsulate this transaction bookkeeping; it would be really easy to forget something. Maybe something that takes a lambda?